### DISCUSSION TITLE
Grid layout with auto-adjusted sizes would be amazing in Reveal

### DISCUSSION THREAD

**Discussion by kjohnsen** [quarto-dev/quarto-cli#3904] [Feature Requests]
I'm loving making Reveal.js presentations with Quarto. However, I struggle to get sizes right when I want to use a grid layout. I've been using the [figure panel](https://quarto.org/docs/authoring/figures.html#complex-layouts) classes to make grids, which seems to get image sizes right for one row, but not for more. For example, 
```qmd
:::{layout="[[1,1],[1]]"
![](img1.jpg)
![](img2.jpg)
![](img3.jpg)
:::
```
often produces output that doesn't fit on the page (that is, img3 is down below, running off the slide). It looks like width of each subpanel is matched well but not height. Having this grid layout automatically size contents to fit on the slide would be really great

**Comment by mcanouil** [comment]
Related to:
- #6166

To note, this is tricky because of the way Reveal.js builds the components.

One workaround is to play with the following CSS rule to constrain the maximum absolute height that your images can take.

```css
.quarto-layout-cell img {
  max-height: 300px;
}
```
